File: toy-problems/Mcdowell-2016/ch.01/question1.1/src/test/java/SolutionTest.java
import static org.junit.Assert.*;
import org.junit.Test;

public class SolutionTest {

    @Test
    public void testIsUniqueChars() {
         assertTrue( Solution.isUniqueChars("uniq") );
        assertFalse( Solution.isUniqueChars("non-unique") );
    }

}

File: toy-problems/Mcdowell-2016/ch.01/question1.1/src/main/go/solution_test.go
package solution 

import "testing"

func Test(t *testing.T) {
	var testCases = []struct {
		input string
                output bool
	}{
		{"uniq", true},
		{"non-unique", false},
	}
	for _, c := range testCases {
		got := HasAllUniqueRunes(c.input)
		if got != c.output {
			t.Errorf("HasAllUniqueRunes(%q) === %t, expected %t", c.input, got, c.output)
		}
	}
}
